snomed-ct-integration/src/test/java/dev/ikm/tinkar/snomedct/integration/SnomedAxiomSemanticIT.java

```diff
@@ -0,0 +1,117 @@
+import dev.ikm.tinkar.entity.EntityService;
+import dev.ikm.tinkar.entity.PatternEntityVersion;
+import dev.ikm.tinkar.entity.SemanticRecord;
+import dev.ikm.tinkar.entity.SemanticVersionRecord;
+import dev.ikm.tinkar.terms.EntityProxy;
+import dev.ikm.tinkar.terms.TinkarTerm;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SnomedAxiomSemanticIT {
+
+    @BeforeAll
+    public static void setup() {
+        CachingService.clearAll();
+        File datastore = new File(System.getProperty("user.home") + "/Solor/September2024_ConnectathonDataset_v1");
+//        File datastore = new File(System.getProperty("user.home") + "/Solor/snomedct-international_AMIA_Nov_2024");
+        ServiceProperties.set(ServiceKeys.DATA_STORE_ROOT, datastore);
+        PrimitiveData.selectControllerByName("Open SpinedArrayStore");
+        PrimitiveData.start();
+    }
+
+    @AfterAll
+    public static void shutdown() {
+        PrimitiveData.stop();
+    }
+
+    /**
+     * Test Snomed Axiom Semantics.
+     *
+     * @result Reads content from file and validates Snomed Axiom of Semantics by calling private method assertSnomedAxioms().
+     */
+    @Test
+    public void testSnomedAxiomSemantics() throws IOException {
+        // Given
+        String sourceFilePath = System.getProperty("user.home") + "/data/SnomedCT_InternationalRF2_PRODUCTION_20241001T120000Z/Full/Terminology/sct2_sRefset_OWLExpressionFull_INT_20241001.txt";
+        String errorFile = "target/failsafe-reports/snomedaxioms_not_found.txt";
+        int notFound = 0;
+        // When
+        try (BufferedReader br = new BufferedReader(new FileReader(sourceFilePath));
+             BufferedWriter bw = new BufferedWriter(new FileWriter(errorFile))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                if (line.startsWith("id")) continue;
+                String[] columns = line.split("\\t");
+
+                //pass these args in assertion method
+                long effectiveTime = SnomedUtility.snomedTimestampToEpochSeconds(columns[1]);
+                StateSet snomedAxiomStatus = Integer.parseInt(columns[2]) == 1 ? StateSet.ACTIVE : StateSet.INACTIVE;
+                EntityProxy.Concept moduleId = EntityProxy.Concept.make(PublicIds.of(UuidUtil.fromSNOMED(columns[3])));
+                EntityProxy.Concept refsetId = EntityProxy.Concept.make(PublicIds.of(UuidUtil.fromSNOMED(columns[4])));
+//                UUID id = UuidUtil.fromSNOMED(columns[0]);
+//                UUID id = UuidT5Generator.get(UUID.fromString("3094dbd1-60cf-44a6-92e3-0bb32ca4d3de"), columns[0]);
```

Comment:
This commented line is the one you want to use to generate the id. Also, older datasets may not work. You'll want to use the dataset that is generated by the rest of the transformation pipeline. For the other tests we've been using System.getProperty("user.home") + "/Solor/generated-data". Just make sure it matches whatever is set in "<datastorePath>" in the pipeline pom.xml.

Reply:
Changed applied.